packages/framework/src/Component/Image/ImageFactory.php

```diff
@@ -81,8 +86,8 @@ public function createMultiple(
         }
 
         $images = [];
-        foreach ($temporaryFilenames as $temporaryFilename) {
-            $images[] = $this->create($imageEntityConfig->getEntityName(), $entityId, $type, $temporaryFilename);
+        foreach ($temporaryFilenames as $key => $temporaryFilename) {
+            $images[] = $this->create($imageEntityConfig->getEntityName(), $entityId, $names[$key] ?? [], $temporaryFilename, $type);
```

Comment:
I am wondering if the coalescence makes sense and if such a case can occur, if it is not some kind of programming error. It would then be hidden by that coalescence. Wouldn't it be better to throw the exception?

Reply:
That array can be empty, I do not want to force it so I solved it like this, feel free to suggest better solution.